iothub/device/src/DeviceClient.cs

```diff
@@ -476,7 +476,21 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// </summary>
         /// <param name="message">The message to send. Should be disposed after sending.</param>
         /// <exception cref="ArgumentNullException">Thrown when a required parameter is null.</exception>
-        /// <exception cref="TimeoutException">Thrown when the operation has been canceled or timed out.</exception>
+        /// <exception cref="TimeoutException">Thrown if the service does not respond to the request within the timeout specified for the operation.
+        /// The timeout values are largely transport protocol specific. Check the corresponding transport settings to see if they can be configured.
+        /// The operation timeout for the client can be set using <see cref="OperationTimeoutInMilliseconds"/>.</exception>
+        /// <exception cref="IotHubCommunicationException">Thrown if the client encounters a transient retryable exception. </exception>
+        /// <exception cref="SocketException">Thrown if a socket error occurs.</exception>
+        /// <exception cref="WebSocketException">Thrown if an error occurs when performing an operation on a WebSocket connection.</exception>
+        /// <exception cref="IOException">Thrown if an I/O error occurs.</exception>
+        /// <exception cref="ClosedChannelException">Thrown if the MQTT transport layer closes unexpectedly.</exception>
+        /// <exception cref="IotHubException">Thrown if an error occurs when communicating with IoT Hub service.
+        /// If <see cref="IotHubException.IsTransient"/> is set to <c>true</c> then it is a transient exception.
+        /// If <see cref="IotHubException.IsTransient"/> is set to <c>false</c> then it is a non-transient exception.</exception>
+        /// <remarks>
+        /// In case of a transient issue, retrying the operation should work. In case of a non-transient issue, inspect the error details and take steps accordingly.
+        /// Please note that the above list is not exhaustive.
```

Comment:
I realize this text may have been supplied to you by Abhipsa, but I have a small issue with it.

It doesn't specify what list is not exhaustive, and when translated to the web documentation the list of exceptions are likely below the remarks section IIRC.

I suggest we reference the list directly.

```suggestion
        /// Please note that the list of exceptions is not exhaustive.
```

If you'd apply this change (or feel free to improve upon it) to this and all the other cases in this PR, that would be ideal.